### PR TITLE
Adding Physical Switch power action

### PR DESCRIPTION
**Pull request by douglasgabriel**
__This PR is able to__
- Add the `restart` action to the `/physical_switches` endpoint

__Depends on__
~https://github.com/ManageIQ/manageiq-providers-lenovo/pull/177~ - Merged
~https://github.com/ManageIQ/manageiq/pull/17548~ - Merged

### CODE REVIEW

**Comment by douglasgabriel** [comment]
I'm closing and reopen this PR just to re-execute the Travis :+1:

**Comment by douglasgabriel** [comment]
@miq-bot assign @imtayadeway

**Comment by agrare** [comment]
@miq-bot assign @gtanzillo

**Comment by gtanzillo** [comment]
@abellotti Can you take a look at this one and let me know if the look ok

File: app/controllers/api/physical_switches_controller.rb [hunk 1]
```diff
@@ -12,8 +12,59 @@ def refresh_resource(type, id, _data = nil)
...
+        rescue StandardError => err
+          action_result(false, err.to_s)
+        end
+      end
+    end
+
+    #
+    # Enqueues the action for multiple resources.
+    #   For multiple resources, when an error occurs, the error messages must
+    #   be built individually for each resource. Always responding with status 200.
```
Comment (gtanzillo) [inline app/controllers/api/physical_switches_controller.rb]: @douglasgabrieIt's not obvious what the difference is between `enqueue_action_single_resource` and `enqueue_action_multiple_resources`. Can you provide some more details?

Comment (douglasgabriel) [reply app/controllers/api/physical_switches_controller.rb]: Sure! Basically a single resource request failure will look like:
```
[status = 404]
{
    "error": {
        "kind": "not_found",
        "message": "Couldn't find PhysicalSwitch with 'id'=4 [WHERE \"switches\".\"type\" IN ('PhysicalSwitch', 'ManageIQ::Providers::Lenovo::PhysicalInfraManager::PhysicalSwitch')]",
        "klass": "ActiveRecord::RecordNotFound"
    }
}
```
However, a multiple resource request failure will look like:
```
[status = 200]
{
    "results": [
        {
            "success": true,
            "message": "Performing restart for Physical Switch id:1 name: 'Lenovo Switch'",
            "task_id": "2",
            "task_href": "http://localhost:3000/api/tasks/2",
            "href": "http://localhost:3000/api/physical_switches/1"
        },
        {
            "success": false,
            "message": "Couldn't find PhysicalSwitch with 'id'=9999 [WHERE \"switches\".\"type\" IN ('PhysicalSwitch', 'ManageIQ::Providers::Lenovo::PhysicalInfraManager::PhysicalSwitch')]"
        }
    ]
}
```
The main difference here is that for a single resource request, the entire response fails, see the status.

But, for multiple resources, the failures are self contained, once some resources could execute the action successfully and others could fail, see the status again.

This is why the `enqueue_action_multiple_resources` method has a `rescue` for these errors and inside it a graceful message is mounted for the user, which doesn't occurs in `enqueue_action_single_resource`.

It made the things more clear for you @gtanzillo ? Maybe add those examples in the method documentation would be a good option, what do you think?

Comment (douglasgabriel) [reply app/controllers/api/physical_switches_controller.rb]: One at time

**Pull request merged**